### ISSUE TITLE
Accessibility issue with keyboard focus indicator on steps grid

### ISSUE THREAD

**Issue by bbsmooth** [freeCodeCamp/freeCodeCamp#50023]
### Describe the Issue

Since we recently added a global focus outline color, the keyboard focus state on the New RWD steps grid now does not have a large enough color contrast to be accessible when focused on a completed step. This is an issue in both light and dark mode. In the following images, assume step 3 is completed.

![fcc-steps1](https://user-images.githubusercontent.com/5926192/230783357-ba125757-b742-40bd-ab7e-05921e104b09.png)

Neither the blue outline or the light gray background has enough contrast with the original lighter blue background they are replacing to be accessible under WCAG.

![fcc-steps1-dark](https://user-images.githubusercontent.com/5926192/230784675-e0ee22de-eec3-47c0-8c2f-8fccaddc2b4e.png)

Again, neither the blue outline or the dark background has enough contrast with the original dark blue background they are replacing to meet WCAG contrast requirements.

I didn't want to make a unilateral decision on how to fix this, so here are a few options to consider:

1. Move the outline so it is just outside of the step square:

![fcc-step2](https://user-images.githubusercontent.com/5926192/230783682-1dbc4476-2f52-4552-8609-fe04f7ad9380.png)

This is accessible because the blue outline now has enough contrast with the white background it is replacing and as an added bonus, the focus ring makes the square appear bigger. Technically, we do not need to have that thin strip of white space between the square and the outline. The outline could be right up against the edge of the square. I like the white space because I think it makes the outline stand out a little more and makes the square look even bigger. This technique will also work in dark mode for the same reasons.

![fcc-step2-dark](https://user-images.githubusercontent.com/5926192/230785287-47cb79a8-4aea-4a30-a6cf-42c193a1b8b2.png)

2. Change the color of the focus outline so it has more contrast. Perhaps the same color as the text?

![fcc-steps3](https://user-images.githubusercontent.com/5926192/230784101-06e0fa61-966f-44d8-b3c2-c465022d8060.png)

![fcc-steps3-dark](https://user-images.githubusercontent.com/5926192/230785108-439bbf94-0d3c-4d13-8359-46ce99cfa6c8.png)

3. I'm not that creative, so please feel free to suggest something else if.

Also, while we are fixing this, I would recommend we increase the font weight from `700` to `900` and the font size from `1.3rem` to `1.5rem` on the completed steps to make the difference in those numbers stand out a little more since we can't rely on the background color for the completed steps to convey their state since they don't have enough color contrast with the background.

### Affected Page

Any steps page

### Steps to Reproduce

N/A

### Expected behavior

Completed steps should have a keyboard focus indicator that meets WCAG standards.

### System

N/A

**Comment by nyni123** [comment]
Hi there!
Thank you for suggesting some possible solutions to address the accessibility issue with the keyboard focus indicator on the steps grid. 

I wanted to add another option to consider, which is to adjust the color of the completed steps background to provide a better contrast with the focus outline. This could involve using a darker or lighter shade of blue that meets WCAG contrast requirements while still maintaining the visual integrity of the design. 

Please let me know your thoughts on this suggestion. 

Thank you!

**Comment by nyni123** [comment]
okay I am on it , will update with new PR 

Thank you

**Comment by bbsmooth** [comment]
@Sboonny Don't beat yourself up over this. I just didn't know if we wanted to go with the default blue indicator or change them to match the text color.

**Comment by a2937** [comment]
Hmmm...I'm a huge fan of the second option. Do I have permission to go ahead on a PR for it?

**Comment by Sboonny** [comment]
How close are you @nyni123 to finishing it, and do you need help with the issue? 

If you would like to fix this issue, please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/), we prioritize contributors following the instructions in our guides. Join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing, our prolific contributors will guide you through this.

> Hmmm...I'm a huge fan of the second option. Do I have permission to go ahead on a PR for it?

Yeab. Although in an ideal world, the issues can be solved by cooperation through campers, which will clear more issues and give a better experience for both, it's rarely the case 😞, so don't worry if you can't reach each other and communicate through the issue.

**Comment by Vansh-Baghel** [comment]
@nyni123 Are you still working on this issue?

**Comment by Vansh-Baghel** [comment]
@bbsmooth Where could I find the affected page or file name is possible?

**Comment by harmeetsingh01** [comment]
@Sboonny , this issue is still open right? Thinking of implementing the 2nd approach described above.

**Comment by Sboonny** [comment]
yeab, go for it

**Comment by Vansh-Baghel** [comment]
Which is the file where changes are to be made?

**Comment by Sboonny** [comment]
https://github.com/freeCodeCamp/freeCodeCamp/blob/ee5fae5d1002d1c5e541e743daaa5081279643eb/client/src/templates/Introduction/components/challenges.tsx#L78

https://github.com/freeCodeCamp/freeCodeCamp/blob/ee5fae5d1002d1c5e541e743daaa5081279643eb/client/src/templates/Introduction/intro.css#L270

**Comment by Vansh-Baghel** [comment]
Cool, starting the work 🫡
